File: python/7 kyu/357.py
# https://www.codewars.com/kata/5a4d303f880385399b000001/train/python

# Passed

def strong_num(number):
    def calculate_factorial(num):
        result = 1
        for i in range(1, num + 1):
            result *= i
        
        return result

    total_sum = sum(calculate_factorial(int(digit)) for digit in str(number))
    is_strong_num = total_sum == number
    
    result = "STRONG!!!!" if is_strong_num else "Not Strong !!"
    return result

output = strong_num(145)
print(output)

File: nodejs/7 kyu/357.js
// https://www.codewars.com/kata/5c942f40bc4575001a3ea7ec/train

// Passed

function per(n) {
  let getDigits = (num) =>
    String(num)
      .split("")
      .map((strDigit) => Number(strDigit));
  let digits = getDigits(n);
  let result = [];
  while (digits.length > 1) {
    n = digits.reduce((prev, curr) => prev * curr, 1);
    digits = getDigits(n);
    result.push(n);
  }

  return result;
}

const output = per(10);
console.log(output);